dataplane/dataplane.cpp

```diff
@@ -1672,6 +1679,124 @@ eResult cDataPlane::parseJsonPorts(const nlohmann::json& json)
 	return eResult::success;
 }
 
+std::optional<std::map<tCoreId, std::set<InterfaceName>>> cDataPlane::parseControlPlaneWorkers(const nlohmann::json& root)
+{
+	nlohmann::json dflt;
+	auto cpw = root.find("controlPlaneWorkers");
+	if (cpw == root.end())
+	{
+		dflt = makeLegacyControlPlaneWorkerConfig();
+		YADECAP_LOG_WARNING("no config for control plane workers provided, using default legacy config\"%s\"\n",
+		                    nlohmann::to_string(dflt).c_str());
+		cpw = dflt.find("controlPlaneWorkers");
+	}
+
+	std::map<tCoreId, std::set<InterfaceName>> result;
+
+	auto add_worker = [&](const nlohmann::json& j) {
+		auto worker = parseControlPlaneWorker(j);
+		if (!worker)
+		{
+			YADECAP_LOG_ERROR("invalid control plane worker config \"%s\"\n", nlohmann::to_string(j).c_str());
+			return false;
+		}
+		result[worker->first] = worker->second;
+		return true;
+	};
+
+	if (!cpw->is_array())
+	{
+		if (!add_worker(cpw.value()))
+		{
+			return std::nullopt;
+		}
+	}
+	else
+	{
+		for (auto j : cpw.value())
+		{
+			if (!add_worker(j))
+			{
+				return std::nullopt;
+			}
+		}
+	}
+	return std::optional{std::move(result)};
+}
+
+nlohmann::json cDataPlane::makeLegacyControlPlaneWorkerConfig()
+{
+	nlohmann::json j;
+	j["core"] = config.controlPlaneCoreId;
+	j["interfaces"] = workerInterfacesToService();
+	return nlohmann::json{{"controlPlaneWorkers", j}};
+}
+
+std::set<InterfaceName> cDataPlane::workerInterfacesToService()
+{
+	std::set<InterfaceName> res;
+	for (const auto& p : config.workers)
+	{
+		const auto& ifaces = p.second;
+		for (const auto& i : ifaces)
+		{
+			res.emplace(i);
+		}
+	}
+	return res;
```

Comment:
I think
```cpp
        std::set<InterfaceName> res;
        for (const auto& [_, ifaces] : config.workers) {
             res.insert(ifaces.begin(), ifaces.end());
        }
        return res;
```
is better

Reply:
Yea, `_` is shown as unused which is a gcc bug, unfortunately.

But `insert` is still desirable though